Sources/EasyPurchase/FetchProduct.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  FetchProduct.swift
+//  EasyPurchase
+//
+//  Created by Sadat Ahmed on 19.09.2023.
+//
+
+import Foundation
+import StoreKit
+
+// Protocol defining actions that can be performed on an in-app product request
+protocol InAppRequestActions: AnyObject {
+    func start()
+    func cancel()
+}
+
+// Protocol extending InAppRequestActions and adding additional properties
+protocol InAppProductRequest: InAppRequestActions {
+    var isCompleted: Bool { get }
+    var cachedProducts: Products? { get }
+}
+
+class FetchProduct : NSObject, SKProductsRequestDelegate, InAppProductRequest {
+    private var productCompletionHandler: ProductComplitionHandler?
+    private var productRequest: SKProductsRequest?
+
+    // Boolean to track if the request is completed
+    var isCompleted: Bool = false
+
+    // Cached products result
+    var cachedProducts: Products?
+
+    init(productIds: Set<String>, productComplitionHandler: @escaping ProductComplitionHandler) {
```

Comment:
ProductComplitionHandler
-> I don't find its definition here. It maybe a type alias. It will be helpful if I see it.

Reply:
@RejaulHasan san, It's in the `InAppProductFetcherBuilder` class.